build_runner/lib/src/entrypoint/test.dart

```diff
@@ -154,6 +157,21 @@ void _ensureBuildTestDependency(PackageGraph packageGraph) {
   }
 }
 
+void _ensureProcessExit(Process process) {
+  var signalsSub = _exitProcessSignals.listen((signal) async {
+    await process.exitCode;
```

Comment:
nit: consider logging something to stdout here `stdout.writeln('waiting for subprocess to exit...');`